gpt_json/tests/test_streaming.py

```diff
@@ -0,0 +1,268 @@
+        (EXAMPLE_LIST_STR_DICT, EXAMPLE_LIST_STR_DICT_STREAM_DATA),
+        (EXAMPLE_DICT, EXAMPLE_DICT_STREAM_DATA),
+        (EXAMPLE_STR_LIST, EXAMPLE_STR_LIST_STREAM_DATA),
+        (EXAMPLE_MULTI_NESTED, EXAMPLE_MULTI_NESTED_STREAM_DATA)
+    ]
+)
+def test_get_expected_stream_objs(full_object, expected_stream_data):
+    out = list(_get_expected_stream_partial_objs(full_object))
+    assert out == expected_stream_data
+
+
+MOCK_ASSISTANT_CHUNK = {"id": "chatcmpl-7GWTw9HlmVFOiXyWNBfNKVFzA55yy", "object": "chat.completion.chunk", "created": 1684172464, "model": "gpt-4-0314", "choices": [{"delta": {"role": "assistant"}, "index": 0, "finish_reason": None}]}
+MOCK_FINISH_REASON_CHUNK = {"id": "chatcmpl-7GWTw9HlmVFOiXyWNBfNKVFzA55yy", "object": "chat.completion.chunk", "created": 1684172464, "model": "gpt-4-0314", "choices": [{"delta": {}, "index": 0, "finish_reason": "stop" }]}
+MOCK_CONTENT_CHUNK = lambda content: {"id": "chatcmpl-7GWTw9HlmVFOiXyWNBfNKVFzA55yy", "object": "chat.completion.chunk", "created": 1684172464, "model": "gpt-4-0314", "choices": [{"delta": {"content": content}, "index": 0, "finish_reason": None}]}
+
+def _mock_oai_streaming_chunks(full_object, json_indent_level=2, prefix_str=None, postfix_str=None):
+    yield MOCK_ASSISTANT_CHUNK
+
+    full_content = f"{prefix_str if prefix_str else ''}{json.dumps(full_object, indent=json_indent_level)}{postfix_str if postfix_str else ''}"
+    full_content_tokens = _tokenize(full_content)
+
+    for token in full_content_tokens:
+        yield MOCK_CONTENT_CHUNK(token)
+    
+    yield MOCK_FINISH_REASON_CHUNK
+    
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "full_object,schema_typehint,expected_stream_data,should_support",
+    [
+        (EXAMPLE_STR_DICT, ExampleStrDictSchema, EXAMPLE_STR_DICT_STREAM_DATA, True),
+        # TODO: support these cases in v1
+        (EXAMPLE_DICT, ExampleDictSchema, EXAMPLE_DICT_STREAM_DATA, False),
+        (EXAMPLE_STR_LIST, ExampleStrListSchema, EXAMPLE_STR_LIST_STREAM_DATA, False),
+        (EXAMPLE_MULTI_NESTED, ExampleMultiNestedSchema, EXAMPLE_MULTI_NESTED_STREAM_DATA, False)
+    ]
+)
+async def test_gpt_stream(full_object, schema_typehint, expected_stream_data, should_support):
+    model_version = GPTModelVersion.GPT_3_5
+    messages = [
+        GPTMessage(
+            role=GPTMessageRole.USER,
+            content="Input prompt",
+        )
+    ]
+
+    model = GPTJSON[schema_typehint](
+        None,
+        model=model_version,
+        temperature=0.0,
+        timeout=60,
+    )
+
+    # Define mock response
+    mocked_oai_raw_responses = _mock_oai_streaming_chunks(full_object)
+    async def async_list_to_generator(my_list):
+        for item in my_list:
+            yield item
+    mock_response = async_list_to_generator(mocked_oai_raw_responses)
+
+    # Create the mock
+    with patch.object(openai.ChatCompletion, "acreate", return_value=mock_response) as mock_acreate:
```

Comment:
I rewrote some of these patches as part of a PR today; since Python got to 3.10 or so they support these async mocks natively:
https://github.com/piercefreeman/gpt-json/pull/11/files#diff-332ec0419d31fe7c72492cf327fc8f968e7122b763badaccba0821c8118cc09dR153